source/Engine.cpp

```diff
@@ -2686,19 +2693,20 @@ void Engine::CreateStatusOverlays()
 			continue;
 
 		if(it == flagship)
-			EmplaceStatusOverlay(it, overlaySettings[Preferences::OverlayType::FLAGSHIP], 0);
+			EmplaceStatusOverlay(it, overlaySettings[Preferences::OverlayType::FLAGSHIP], 0, it->Cloaking());
 		else if(it->GetGovernment()->IsEnemy())
-			EmplaceStatusOverlay(it, overlaySettings[Preferences::OverlayType::ENEMY], 2);
+			EmplaceStatusOverlay(it, overlaySettings[Preferences::OverlayType::ENEMY], 2, it->Cloaking());
 		else if(it->IsYours() || it->GetPersonality().IsEscort())
-			EmplaceStatusOverlay(it, overlaySettings[Preferences::OverlayType::ESCORT], 1);
+			EmplaceStatusOverlay(it, overlaySettings[Preferences::OverlayType::ESCORT], 1, it->Cloaking());
```

Comment:
I understand why we make the status overlay disappear for ships that aren't your own (because the ship itself disappears), but since you can still see your own ships when they're cloaked, does it really make sense to hide the status overlay? Perhaps the if statement at the top of this loop should have `(!it->IsYours() && it->Cloaking() == 1.)` instead of just checking the cloak progress.

Reply:
> I understand why we make the status overlay disappear for ships that aren't your own (because the ship itself disappears), but since you can still see your own ships when they're cloaked, does it really make sense to hide the status overlay? Perhaps the if statement at the top of this loop should have `(!it->IsYours() && it->Cloaking() == 1.)` instead of just checking the cloak progress.

The previous behaviour was to just blink out all status overlays when cloaking hit 100%, but I could do something like what you describe.

Reply:
Yeah, I'm aware that you haven't actually changed the behavior of the status overlay with cloaked escorts. Just musing on whether we should change it.